SonarQube.Client/Models/ServerSentEvents/SSEStreamWriter.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using System.Threading.Channels;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarQube.Client.Models.ServerSentEvents.ServerContract;
+
+namespace SonarQube.Client.Models.ServerSentEvents
+{
+    public interface ISSEStreamWriter : IDisposable
+    {
+        /// <summary>
+        /// Begin pumping events. Will block the calling thread with an infinite loop.
+        /// </summary>
+        Task BeginListening();
```

Comment:
This is the half that writes ISqServerEvent into the channel, for ISSEStreamReader to read, but the method's name doesn't follow the convention. Any suggestions? :S

Reply:
It seems weird to have a `writer` that listens. I don't know what I'd call it instead, though.
Calling the interface `ISSEListener` messes up the reader/writing name.

Could suffix the method name with `Async`.